realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/SchemaTests.kt

```diff
@@ -0,0 +1,195 @@
+        }
+        assertTrue(realm.isClosed)
+    }
+
+    @Test
+    fun createObject() {
+        Realm.getInstance(config).use { realm ->
+            realm.beginTransaction()
+            assertTrue(realm.schema.contains("StringOnly"))
+            val stringOnly = realm.createObject(StringOnly::class.java)
+            stringOnly.chars = "TEST"
+            realm.commitTransaction()
+            assertEquals(1, realm.where(StringOnly::class.java).count())
+        }
+    }
+
+    @Test
+    fun allow_createClass() {
+        DynamicRealm.getInstance(config).use { realm ->
+            val className = "Dogplace"
+            realm.beginTransaction()
+            realm.schema.create("Dogplace")
+            realm.commitTransaction()
+            assertTrue(realm.schema.contains(className))
+        }
+    }
+
+    @Test
+    fun allow_addField() {
+        // Init schema
+        Realm.getInstance(config).close()
+        val className = "StringOnly"
+        DynamicRealm.getInstance(config).use { realm ->
+            val objectSchema = realm.schema[className]
+            assertNotNull(objectSchema)
+            realm.beginTransaction()
+            objectSchema!!.addField("foo", String::class.java)
+            assertTrue(objectSchema.hasField("foo"))
+            realm.commitTransaction()
+            assertTrue(objectSchema.hasField("foo"))
+        }
+    }
+
+    // Special column "__OID" should be hidden from users.
+    @Test
+    fun fieldNames_stableIdColumnShouldBeHidden() {
+        val className = "StringOnly"
+        Realm.getInstance(config).use { realm ->
+            val objectSchema = realm.schema[className]
+            assertNotNull(objectSchema)
+            val names = objectSchema!!.fieldNames
+            assertEquals(1, names.size)
+            assertEquals(StringOnly.FIELD_CHARS, names.iterator().next())
+        }
+    }
+
+    enum class DestructiveSchemaOperation {
+        REMOVE_CLASS,
+        RENAME_CLASS,
+        SET_CLASS_NAME,
+        REMOVE_FIELD,
+        RENAME_FIELD,
+        REMOVE_INDEX,
+        REMOVE_PRIMARY_KEY,
+        ADD_PRIMARY_KEY,
+        ADD_FIELD_PRIMARY_KEY,
+    }
+
+    @Test
+    fun disallowDestructiveUpdateOfSyncedRealm() {
+        for (operation in DestructiveSchemaOperation.values()) {
+            // Init schema
+            Realm.getInstance(config).close()
+            val className = "StringOnly"
+            val newClassName = "Dogplace"
+            val fieldName = "chars"
+            val newFieldName = "newchars"
+
+            DynamicRealm.getInstance(config).use { realm ->
+                assertTrue(realm.schema.contains(className))
+                val objectSchema = realm.schema[className]!!
+                assertNotNull(objectSchema)
+                assertTrue(objectSchema.hasField(fieldName))
+
+                realm.beginTransaction()
+                errorCollector.assertFailsWith<UnsupportedOperationException> {
+                    when (operation) {
```

Comment:
Normally we have a default case that always fail, to ensure that the test will fail if we add more enums and forget to update tests.